XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs

```diff
@@ -407,111 +407,18 @@ private void InitializeServiceLocator()
 
             ClearCoreSystemCache();
 
-            #region Services Registration
-
-            if (ActiveProfile.IsCameraSystemEnabled)
+            foreach (var configuration in ActiveProfile.RegisteredServiceConfigurations)
             {
-                if (TryCreateAndRegisterService<IMixedRealityCameraSystem>(ActiveProfile.CameraSystemType, out var service, ActiveProfile.CameraSystemProfile) && CameraSystem != null)
-                {
-                    TryRegisterDataProviderConfigurations(ActiveProfile.CameraSystemProfile.RegisteredServiceConfigurations, service);
-                }
-                else
+                if (configuration.Enabled)
                 {
-                    Debug.LogError("Failed to start the Camera System!");
-                }
-            }
-
-            if (ActiveProfile.IsInputSystemEnabled)
-            {
-                if (TryCreateAndRegisterService<IMixedRealityInputSystem>(ActiveProfile.InputSystemType, out var service, ActiveProfile.InputSystemProfile) && InputSystem != null)
-                {
-                    if (TryCreateAndRegisterService<IMixedRealityFocusProvider>(ActiveProfile.InputSystemProfile.FocusProviderType, out _))
```

Comment:
This one might be tricky to figure out. We do have some special edge cases when we register some of our systems.

@SimonDarksideJ @FejZa